### PR TITLE
Custom Asset Default Display Preferences

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

File: src/Glpi/Asset/AssetDefinition.php [hunk 1]
```diff
@@ -478,12 +478,31 @@ private function prepareInput(array $input): array|bool
...
             // Force menu refresh when active state change
             unset($_SESSION['menu']);
         }
+
+        // Add default display preferences for the new asset definition
+        // Status, Manufacturer, Serial, Type, Model, Location, Last Update
+        $prefs = [31, 23, 5, 4, 40, 3, 19];
+        $rank = 1;
+        foreach ($prefs as $field) {
+            $DB->insert(
```
Comment (cedric-anne) [inline src/Glpi/Asset/AssetDefinition.php:496]: I think it should be done using the `DisplayPreference::add()` method. I know most of operations in `DisplayPreference` class are done directly with low level DB methods, but IMHO, it is an issue as it prevent plugins to be able to use hooks on these operations.

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Tests are failing.

File: src/Glpi/Asset/AssetDefinition.php [hunk 2]
```diff
@@ -478,12 +478,27 @@ private function prepareInput(array $input): array|bool
...
         if ($this->isActive()) {
             $this->syncProfilesRights();
 
             // Force menu refresh when active state change
             unset($_SESSION['menu']);
         }
+
+        // Add default display preferences for the new asset definition
+        // Status, Manufacturer, Serial, Type, Model, Location, Last Update
+        $prefs = [31, 23, 5, 4, 40, 3, 19];
```
Comment (cedric-anne) [inline src/Glpi/Asset/AssetDefinition.php:493]: I just reorder preferences to match the default computer preferences.

```suggestion
        $prefs = [
            4, // Name
            40, // Model
            5, // Serial
            23, // Manufacturer
            31, // Status
            3, // Location
            19, // Last Update
        ];
```

**Review by cedric-anne — APPROVED** [review]

File: src/Glpi/Asset/AssetDefinition.php [hunk 3]
```diff
@@ -478,12 +478,34 @@ private function prepareInput(array $input): array|bool
 
     public function post_addItem()
     {
+        /** @var \DBmysql $DB */
+        global $DB;
+
```
Comment (cedric-anne) [inline src/Glpi/Asset/AssetDefinition.php:483]:
```suggestion
```

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**